src/function/table/arrow_conversion.cpp

```diff
@@ -166,20 +166,25 @@ static ArrowListOffsetData ConvertArrowListViewOffsetsTemplated(Vector &vector,
 	// ListViewArrays do not have this same constraint
 	// for that reason we need to keep track of the lowest offset, so we can skip all the data that comes before it
 	// when we scan the child data
-	auto lowest_offset = offsets[0];
+
+	auto lowest_offset = size ? offsets[0] : 0;
 	auto list_data = FlatVector::GetData<list_entry_t>(vector);
 	for (idx_t i = 0; i < size; i++) {
 		auto &le = list_data[i];
 		le.offset = offsets[i];
 		le.length = sizes[i];
 		list_size += le.length;
-		lowest_offset = MinValue(lowest_offset, offsets[i]);
+		if (sizes[i] != 0) {
+			lowest_offset = MinValue(lowest_offset, offsets[i]);
+		}
 	}
 	start_offset = lowest_offset;
-	// We start scanning the child data at the 'start_offset' so we need to fix up the created list entries
-	for (idx_t i = 0; i < size; i++) {
-		auto &le = list_data[i];
-		le.offset -= start_offset;
+	if (start_offset) {
+		// We start scanning the child data at the 'start_offset' so we need to fix up the created list entries
+		for (idx_t i = 0; i < size; i++) {
+			auto &le = list_data[i];
+			le.offset -= start_offset;
```

Comment:
if you keep the `sizes[i] != 0` in the `lowest_offset` calculation above, you have to:

1. skip `sizes[i] == 0` entries here as well to not go negative on the offsets
2. set `le.offset` to `0` because if you keep in unchanged it might end up bigger than the trimmed child array
3. even setting `le.offset` to `0` migth not be enough because `0 + sizes[i]` might end up bigger the the trimmed child array

(sorry for bringing up this optimization (in terms of compression of child array) and then pointing out all these complexities it creates)

Reply:
Thanks, that's a bit of an oversight on my part
I don't think 3 is relevant though, we only skip if `size == 0`, so I don't see how the `0 + sizes[i]` can ever be bigger than the child array?

Note that we don't trim both ways, only the start

Reply:
You're right. 3 only matters if you skipped nulls *and* trimmed at the end as well.